src/wizard/wizard-form.tsx

```diff
@@ -52,11 +53,12 @@ export default function WizardFormWithAnalytics(props: WizardFormProps) {
   const __internalRootRef = useComponentMetadata('WizardForm', PACKAGE_VERSION, analyticsMetadata as AnalyticsMetadata);
   const stepHeaderRef = useRef<HTMLDivElement | null>(null);
 
-  useEffectOnUpdate(() => {
-    if (stepHeaderRef && stepHeaderRef.current) {
+  const previousActiveStepIndex = usePrevious(props.activeStepIndex);
+  useEffect(() => {
```

Comment:
React documentation specifically calls such "fix" an anti-pattern: https://react.dev/learn/synchronizing-with-effects#how-to-handle-the-effect-firing-twice-in-development

Thanks for finding React 18 compatibility issues, but we need to fix them properly.

Reply:
I read the React guidelines, too - but I couldn't find a good answer for this particular problem. We don't need any cleanup here, but we do need to ensure the focus() only fires after step change but never on the first render.

Reply:
Thanks, the second version looks better